code/modules/antagonists/creep/creep.dm

```diff
@@ -254,21 +259,28 @@
 
 /datum/objective/polaroid/update_explanation_text()
 	..()
-	if(target && target.current)
+	if(target?.current)
 		explanation_text = "Take a photo of [target.name] while they're alive."
 	else
 		explanation_text = "Free Objective"
 
 /datum/objective/polaroid/check_completion()
 	var/list/datum/mind/owners = get_owners()
-	for(var/datum/mind/M in owners)
-		if(!isliving(M.current))
+	for(var/datum/mind/mind in owners)
+		var/mob/living/current = mind.current
+		if(!istype(current))
 			continue
-		var/list/all_items = M.current.GetAllContents()	//this should get things in cheesewheels, books, etc.
-		for(var/obj/I in all_items) //Check for wanted items
-			if(istype(I, /obj/item/photo))
-				var/obj/item/photo/P = I
-				if(P.picture && (target.current in P.picture.mobs_seen) && !(target.current in P.picture.dead_seen)) //Does the picture exist and is the target in it and is the target not dead
+		for(var/obj/item/photo/photo as anything in current.GetAllContents(/obj/item/photo)) //Check for wanted items
+			var/datum/picture/picture = photo.picture
+			if(!picture)
+				continue
+			var/seen_mind_stat = picture.minds_seen[target]
+			if(!isnull(seen_mind_stat) && seen_mind_stat != DEAD)
+				return TRUE
+			for(var/mob/living/carbon/seen in picture.mobs_seen)
+				if(seen.last_mind != target)
+					continue
+				if(picture.mobs_seen[seen] != DEAD)
 					return TRUE
```

Comment:
I think this is wrong.

1. Kill your target anyway
2. Pop out their brain
3. Revive them with a monkey brain
4. Take a photo

This whole if check seems fine to be removed.

Reply:
This was more meant to handle like, if they got gibbed or something after you took the photo?

Reply:
I think mind_seen is enough.